tests/test_authenticator.py

```diff
@@ -2966,11 +2966,12 @@ def test_basic_auth_temp_token_returns_existing(self):
             token = response.json.get('access_token')
             assert token
 
-            another_response = self.controller.basic_auth_temp_token({}, self._db)
-            assert another_response.status_code == 200
+            with freeze_time(lambda: utc_now() + datetime.timedelta(seconds=59)):
+                another_response = self.controller.basic_auth_temp_token({}, self._db)
+                assert another_response.status_code == 200
```

Comment:
This will work most of the time, but I think it's likely to fail if there's a second boundary between specific lines of code.

Rather than freezing time to a time relative to the present, can you freeze time to a specific second (2022-01-01 00:00:00) before issuing the initial token, and then move it around to specific other seconds?